### PR TITLE
Require Dart 3, update lints

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

File: source_gen/lib/src/utils.dart [hunk 1]
```diff
@@ -64,18 +64,12 @@ String urlOfElement(Element element) => element.kind == ElementKind.DYNAMIC
...
-      return normalizeDartUrl(url);
-    case 'package':
-      return packageToAssetUrl(url);
-    case 'file':
-      return fileToAssetUrl(url);
-    default:
-      return url;
-  }
-}
+Uri normalizeUrl(Uri url) => switch (url.scheme) {
```
Comment (natebosch) [inline source_gen/lib/src/utils.dart:67]: 👍

**Review by natebosch — APPROVED** [review]

File: source_gen/lib/src/utils.dart (hunk 1 above)
Comment (kevmoo) [reply source_gen/lib/src/utils.dart:67]: I can't get over how nice it is!

File: source_gen/lib/src/library.dart [hunk 2]
```diff
@@ -98,10 +98,7 @@ class LibraryReader {
   /// * `asset`
   ///
   /// May throw [ArgumentError] if it is not possible to resolve a path.
-  Uri pathToUrl(dynamic toUrlOrString) {
```
Comment (natebosch) [inline source_gen/lib/src/library.dart:101]: This was statically breaking.

This is worth doing, but we might have to wait for a major version bump, or add a `dart fix` fix for it.

Comment (natebosch) [reply source_gen/lib/src/library.dart:101]: https://github.com/dart-lang/source_gen/pull/673

Comment (kevmoo) [reply source_gen/lib/src/library.dart:101]: annoying!

**Pull request merged**